source/output_formatter.py

```diff
@@ -0,0 +1,39 @@
+from os import fsync
+from typing import List
+
+from interface import OutputFormatterInterface, OutputData
+
+
+class OutputFormatterFasta(OutputFormatterInterface):
+
+    def write_output(self, output_data: List[OutputData]) -> None:
+
+        for output in output_data:
+
+            (read_id, _, _, _, _, basecalled_seq, _) = output
+
+            if len(basecalled_seq) == 0:
+                return
+
+            print(">%s" % read_id, file=self.output_stream)
+            print(basecalled_seq, file=self.output_stream, flush=True)
+
+        fsync(self.output_stream.fileno())
+
+
+class OutputFormatterFastq(OutputFormatterInterface):
+
+    def write_output(self, output_data: List[OutputData]) -> None:
```

Comment:
is it List or Sequence?

Reply:
It is List, could be a Sequence, but in my opinion should be a Tuple. It should be a tuple since returned by TaskExecutor. So it would be enforced by design that those data are immutable.

Reply:
If the function *does not* care about whether it is list, tuple, or sequence, it should take the superset of them, i.e. the Sequence. Especially if you don't see the requirement changing in the future. In fact, you might want to pass output data generator to the function instead of passing a list